include/mbgl/storage/default_file_source.hpp

```diff
@@ -2,9 +2,135 @@
 #define MBGL_STORAGE_DEFAULT_FILE_SOURCE
 
 #include <mbgl/storage/file_source.hpp>
+#include <mbgl/util/geo.hpp>
+
+#include <vector>
 
 namespace mbgl {
 
+/*
+ * An offline region defined by a style URL, geographic bounding box, zoom range, and
+ * device pixel ratio.
+ */
+class OfflineTilePyramidRegionDefinition {
+public:
+    OfflineTilePyramidRegionDefinition(const std::string&, const LatLngBounds&, double, double, float);
+
+    std::string styleURL;
```

Comment:
If I’m not mistaken, the style URL determines which data sources get downloaded. For example, taking the Satellite style offline shouldn’t pull in Mapbox Streets data.